### DISCUSSION TITLE
`$tmpl()` does not accept a variable

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#15068] [Ideas]
**V version: 0.3.0 c48ed52** 
**OS: ArchLinux** 

**What did you do?**
https://github.com/vlang/v/blob/master/doc/docs.md#tmpl-for-embedding-and-parsing-v-template-files

Changing this example to use a variable as follows will result in an error.

```v
fn build() string {
	name := 'Peter'
	age := 25
	numbers := [1, 2, 3]
        file := '1.txt'
	return $tmpl(file)  // pass a variable instead of string
}

fn main() {
	println(build())
}
```

**What did you expect to see?**
No error.
 
**What did you see instead?**
```
bug.v:6:15: error: unexpected name `file`, expecting string
    4 |     numbers := [1, 2, 3]
    5 |     file := '1.txt'
    6 |     return $tmpl(file)
      |                  ~~~~
    7 | }
```

**Comment by spytheman** [comment]
It can not support variables, at least for now.

It is a compile time construct, so it has to be able to resolve where the template is at compile time, not at runtime.
